drivers/filesystems/fastfat/cleanup.c

```diff
@@ -186,11 +188,11 @@ VfatCleanup(
     }
 
     ExAcquireResourceExclusiveLite(&IrpContext->DeviceExt->DirResource, TRUE);
-    Status = VfatCleanupFile(IrpContext);
-    ExReleaseResourceLite(&IrpContext->DeviceExt->DirResource);
+    Deleted = VfatCleanupFile(IrpContext);
+    if (!Deleted) ExReleaseResourceLite(&IrpContext->DeviceExt->DirResource);
```

Comment:
So here, one of the suggestions from the other PR was to just do:
```c
if (!VfatCleanupFile(IrpContext))
    ExReleaseResourceLite(&IrpContext->DeviceExt->DirResource);
```
without the intermediate boolean.

Reply:
I think that using a variable explicitly makes it more clear, what's going on here, otherwise people might think it means success.